src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -2273,20 +2303,161 @@ class Program
 {
     static void Main()
     {
-        void Local(int x)
+        int capture1 = 0;
+        void L1(int x) => Console.Write(x);
+        void L2(int x)
+        {
+            Console.Write(capture1);
+            Console.Write(x);
+        }
+        dynamic L4(int x)
+        {
+            Console.Write(capture1);
+            return x;
+        }
+        Action<int> L5(int x)
         {
             Console.Write(x);
+            return L1;
         }
+
         dynamic val = 2;
-        Local(val);
+        Console.WriteLine();
+        L1(val);
+        L2(val);
+        Console.WriteLine();
+        L2(L4(val));
+        L5(val)(val);
     }
 }
 ";
-            VerifyDiagnostics(source,
-    // (12,9): error CS8098: Cannot invoke the local function 'Local' with dynamic parameters.
-    //         Local(val);
-    Diagnostic(ErrorCode.ERR_DynamicLocalFunctionParameter, "Local(val)").WithArguments("Local").WithLocation(12, 9)
-    );
+            VerifyOutput(source, output: @"202
+00222");
+        }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.Dynamic, CompilerFeature.Params)]
+        public void DynamicArgsAndParams()
+        {
+            var src = @"
+int capture1 = 0;
+void L1(int x, params int[] ys)
+{
+    Console.Write(capture1);
+    Console.Write(x);
+    foreach (var y in ys)
+    {
+        Console.Write(y);
+    }
+}
+
+dynamic val = 2;
+int val2 = 3;
+L1(val, val2);
+L1(val);
+L1(val, val, val);
+";
+            VerifyOutputInMain(src, "023020222", "System");
+        }
+
+        [WorkItem(10708, "https://github.com/dotnet/roslyn/issues/10708")]
+        [CompilerTrait(CompilerFeature.Dynamic, CompilerFeature.Params)]
+        [Fact]
+        public void DynamicArgumentToParams()
+        {
+            var src = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        void L1(int x = 0, params int[] ys) => Console.Write(x);
+
+        dynamic val = 2;
+        L1(val, val);
+        L1(ys: val, x: val);
+        L1(ys: val);
+    }
+}";
+            VerifyDiagnostics(src,
+                // (10,9): error CS8106: Cannot pass argument with dynamic type to params parameter 'ys' of local function 'L1'.
+                //         L1(val, val);
+                Diagnostic(ErrorCode.ERR_DynamicLocalFunctionParamsParameter, "L1(val, val)").WithArguments("ys", "L1").WithLocation(10, 9),
+                // (11,9): error CS8106: Cannot pass argument with dynamic type to params parameter 'ys' of local function 'L1'.
+                //         L1(ys: val, x: val);
+                Diagnostic(ErrorCode.ERR_DynamicLocalFunctionParamsParameter, "L1(ys: val, x: val)").WithArguments("ys", "L1").WithLocation(11, 9),
```

Comment:
That may be so; if so it is a violation of the language specification (I could quote a number of sections, but lets take that offline).